integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/mocks/oslpdevice/MockOslpServer.java

```diff
@@ -180,23 +180,23 @@ public void stop() {
     }
 
     public void resetServer() {
-        this.receivedRequests.clear();
         this.receivedResponses.clear();
-        this.mockResponses.clear();
+        this.devicesContext.clear();
         this.channelHandler.reset();
     }
 
-    public Message waitForRequest(final MessageType requestType) {
+    public Message waitForRequest(final String deviceUID, final MessageType messageType)
+            throws DeviceSimulatorException {
         LOGGER.info(
-                "Waiting for request of message type: {} - receivedRequests: {}, receivedResponses: {}, mockResponses: {}",
-                requestType.name(), this.receivedRequests.size(), this.receivedResponses.size(),
-                this.mockResponses.size());
+                "Device {} is waiting for request of message type: {}, receivedResponses: {}",
+                deviceUID, messageType.name(), this.receivedResponses.size());
 
+        final DeviceState deviceState = this.devicesContext.getDeviceState(deviceUID);
         int count = 0;
-        while (!this.receivedRequests.containsKey(requestType)) {
+        while (!deviceState.hasReceivedRequests(messageType)) {
             try {
                 count++;
-                LOGGER.info("Sleeping 1s " + count + " - waiting for request of message type: " + requestType.name());
+                LOGGER.info("Sleeping 1s {} - Device {} is waiting for a request of message type: {}", count, deviceUID, messageType.name());
                 Thread.sleep(1000);
             } catch (final InterruptedException e) {
                 Assertions.fail("Polling for response interrupted");
```

Comment:
Not from this pull request, but the failure message seems deceptive. The waiting here looks like it is for a received request, not for a response. (The same applies just a few lines below with the next `fail`.)

Reply:
true :-) will change this